### PR TITLE
Refactor with terrafrom and defaultCredential for keyvault sample

### PR DESCRIPTION
**Pull request by backwind1233**
Like this PR #104 , refactor one keyvault sample in the same way.
1. provision azure resources with terrafrom.
2. running with the same experience: follow the same steps to run the sample;
3. make readme follow the same pattern.

Not included in the pr, but willing to do.
1. add a description for RBAC.
2. add a link to spring-cloud-azure docs.
3. add a description for the benefit of defaultCredential(why samples refactored to defaultCredential) 

Explanation about why AAD is not included in the PR.
1. The main goal of this task [Migrated to use DefaultCredential if possible](https://github.com/Azure/azure-sdk-for-java/issues/25652)  is supporting and demonstrating defaultCredential feature, **not** prevision azure resources with terrafrom.
2. The aad's samples are not easy to refactor, about the code and doc, the samples have several scenarios and advanced features.
3. We can refactor aad's sample by provisioning resources with terrafrom.

### CODE REVIEW

File: keyvault/spring-cloud-azure-starter-keyvault-secrets/single-property-source/README.md [hunk 1]
```diff
@@ -1,21 +1,135 @@
...
-1. Read [document about register an application](https://docs.microsoft.com/azure/active-directory/develop/quickstart-register-app), register an application. get the `Application (client) ID`: **${AZURE_CLIENT_ID}**, and create a secret, get the `Client Secret Value`: **${AZURE_CLIENT_SECRET}**.
-2. Read [document about create key vault](https://docs.microsoft.com/en-us/azure/key-vault/general/quick-create-portal), get the `Directory ID`: **${AZURE_TENANT_ID}** and `Vault URI` **${ENDPOINT}**.
-3. Read [document about assign Key Vault access policy](https://docs.microsoft.com/en-us/azure/key-vault/general/assign-access-policy?tabs=azure-portal), assign `Secret Management` template to the client(or principal) we created in step 1.
+## What You Need
 
-## Fill the values in application.yml
-Fill these values in application.yml: **${AZURE_TENANT_ID}**, **${AZURE_CLIENT_ID}**, **${AZURE_CLIENT_SECRET}**, **${ENDPOINT}**.
+- [An Azure subscription](https://azure.microsoft.com/free/)
+- [Terraform](https://www.terraform.io/)
+- [IntelliJ IDEA](https://www.jetbrains.com/idea/download/#section=mac)
```
Comment (saragluna) [inline keyvault/spring-cloud-azure-starter-keyvault-secrets/single-property-source/README.md:14]: same here for the IDE

File: keyvault/spring-cloud-azure-starter-keyvault-secrets/single-property-source/README.md [hunk 2]
```diff
@@ -1,21 +1,135 @@
...
+- [Azure CLI](https://docs.microsoft.com/cli/azure/install-azure-cli)
+- JDK8
+- Maven
+
+## Prevision Azure Resources required to run this sample.
```
Comment (saragluna) [inline keyvault/spring-cloud-azure-starter-keyvault-secrets/single-property-source/README.md:19]: Should we remove the '.' from the header?

File: keyvault/spring-cloud-azure-starter-keyvault-secrets/single-property-source/terraform/outputs.tf [hunk 3]
```diff
@@ -0,0 +1,4 @@
+output "keyvault_uri" {
+  value       = azurerm_key_vault.kv_account.vault_uri
+  description = "The keyvault uri."
```
Comment (saragluna) [inline keyvault/spring-cloud-azure-starter-keyvault-secrets/single-property-source/terraform/outputs.tf:3]: key vault

File: keyvault/spring-cloud-azure-starter-keyvault-secrets/single-property-source/README.md [hunk 4]
```diff
@@ -1,21 +1,135 @@
...
+You can go to [Azure portal](https://ms.portal.azure.com/) in your web browser to check the resources you created.
+
+### Export output to your local Environment
+Running the command below to export environment values:
+
+```shell
+ source ./terraform/setup_env.sh
+```
+
+## Run locally
```
Comment (saragluna) [inline keyvault/spring-cloud-azure-starter-keyvault-secrets/single-property-source/README.md:109]: Should we change to Run Locally here?

**Review by saragluna — CHANGES_REQUESTED** [review]

File: keyvault/spring-cloud-azure-starter-keyvault-secrets/single-property-source/README.md (hunk 4 above)
Comment (backwind1233) [reply keyvault/spring-cloud-azure-starter-keyvault-secrets/single-property-source/README.md:109]: maybe, we will add something like # run in asc, that's why ;
for now, it's ok to remove it.

**Review by saragluna — APPROVED** [review]
LGTM.

File: keyvault/spring-cloud-azure-starter-keyvault-secrets/single-property-source/README.md [hunk 5]
```diff
@@ -1,21 +1,136 @@
+# Using Key Vault Secrets With Spring Cloud Azure Key Vault Secrets Starter
 
-# Spring Cloud Azure Starter Key Vault Secrets Sample
+This sample illustrates the simplest usage of `spring-cloud-azure-starter-keyvault-secrets`. 
+To learn all features, please refer to [reference doc](https://microsoft.github.io/spring-cloud-azure/docs/current/reference/html/index.html#secret-management).
```
Comment (saragluna) [inline keyvault/spring-cloud-azure-starter-keyvault-secrets/single-property-source/README.md:4]: Using this symbolic link, current, will make it possible to be broken in the feature. But seems like we don't have a better choice now.

**Pull request merged**